pkg/agent/core/ngt/usecase/agentd.go

```diff
@@ -46,7 +47,28 @@ type run struct {
 }
 
 func New(cfg *config.Data) (r runner.Runner, err error) {
-	ngt, err := service.New(cfg.NGT)
+	ngt, err := service.New(
+		service.WithErrGroup(errgroup.Get()),
+		service.WithEnableInMemoryMode(cfg.NGT.EnableInMemoryMode),
+		service.WithIndexPath(cfg.NGT.IndexPath),
+		service.WithAutoIndexCheckDuration(cfg.NGT.AutoIndexCheckDuration),
+		service.WithAutoIndexDurationLimit(cfg.NGT.AutoIndexDurationLimit),
+		service.WithAutoSaveIndexDuration(cfg.NGT.AutoSaveIndexDuration),
+		service.WithAutoIndexLength(cfg.NGT.AutoIndexLength),
+		service.WithInitialDelayMaxDuration(cfg.NGT.InitialDelayMaxDuration),
+		service.WithMinLoadIndexTimeout(cfg.NGT.MinLoadIndexTimeout),
+		service.WithMaxLoadIndexTimeout(cfg.NGT.MaxLoadIndexTimeout),
+		service.WithLoadIndexTimeoutFactor(cfg.NGT.LoadIndexTimeoutFactor),
+		service.WithNGTOpts(
+			ngt.WithDimension(cfg.NGT.Dimension),
+			ngt.WithDistanceTypeByString(cfg.NGT.DistanceType),
+			ngt.WithObjectTypeByString(cfg.NGT.ObjectType),
+			ngt.WithBulkInsertChunkSize(cfg.NGT.BulkInsertChunkSize),
+			ngt.WithCreationEdgeSize(cfg.NGT.CreationEdgeSize),
+			ngt.WithSearchEdgeSize(cfg.NGT.SearchEdgeSize),
+			ngt.WithDefaultPoolSize(cfg.NGT.DefaultPoolSize),
```

Comment:
I think it's better to set core Option in the service package.
I would be more clear to understand dependency tree

Reply:
i see.
how about to pass the config struct to the service layer like this?
```go
	service.WithNGTConfig(cfg.NGT)
```